src/target.zig

```diff
@@ -644,9 +644,10 @@ pub fn defaultAddressSpace(
         function,
     },
 ) AddressSpace {
-    _ = target;
-    _ = context;
-    return .generic;
+    if (context != .function or target.cpu.arch != .avr) return .generic;
+    // The default address space for functions on AVR is .flash to produce
+    // correct fixups into progmem.
+    return .flash;
```

Comment:
Could you reverse this to make it more readable and maintainable?

Reply:
Sure, done!